pkg/agent/manager/cache/cache.go

```diff
@@ -375,6 +420,40 @@ func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.Regi
 		}
 	}
 
+	// entries with active subscribers which are not cached will be put in staleEntries map
+	activeSubsByEntryID, recordsWithLastAccessTime := c.syncSVIDs()
```

Comment:
yeap, but in this possible scenario I was thinking, 
the issue was with completing the list before we know we are not going to add more entries to that list.

Basically we have limit of `10`, and our agent has `20` entries. our cache has only `7` now,  so we complete the list to 3 more from this call to `syncSVIDs()`.
But after that call we found that we need to get SVIDs for another 2 items, that are not currently in our cache,
so our code will try to get `5` new SVIDs instead of the 2 required + 1 to complete the list of cached SVIDs.
All this in the same sync call.
Makes it sense?

Reply:
> But after that call we found that we need to get SVIDs for another 2 items, that are not currently in our cache
`NewSubscriber` method would be waiting for lock until `UpdateEntries` finishes. Therefore, we won't need to get 2 new SVIDs in middle of `UpdateEntries` execution. And those 2 new subscribers will be serviced in next iteration of sync.